circle.yml

```diff
@@ -1,38 +1,151 @@
-machine:
-  services:
-    - docker
-    - redis
-  node:
-    version:
-      6.9.1
-dependencies:
-  override:
-    - pip install --upgrade setuptools
-    - pip install -r requirements_dev.txt
-    - pip install -r requirements.txt
-    - npm install
-    - npm run build
-  cache_directories:
-    - node_modules/
-test:
-  override:
-    - pytest --junitxml=$CIRCLE_TEST_REPORTS/junit.xml tests/
-deployment:
-  tarball:
-    branch: [master, /release.*/]
-    commands:
-      - bin/pack
-  docker:
-    tag: /v[0-9]+(\.[0-9\-a-z]+)*/
-    commands:
-      - bin/pack
-      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
-      - docker build -t redash/redash:$(./manage.py version | sed -e "s/\+/./") .
-      - docker push redash/redash:$(./manage.py version | sed -e "s/\+/./")
+version: 2.0
 notify:
   webhooks:
     - url: https://webhooks.gitter.im/e/895d09c3165a0913ac2f
-general:
-  branches:
-    ignore:
-      - gh-pages
+jobs:
+  build:
+    docker:
+      - image: circleci/node:6
+    working_directory: ~/redash
+    steps:
+      - checkout
+      - restore_cache:
+         keys:
+           - node-modules-
+      - run:
+          name: Setup NPM Environment
+          command: |
+            npm install
+      - run:
+          name: build
+          command: npm run build
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./*
+      - save_cache:
+         key: node-modules-
+         paths:
+           - node_modules
+  unit_tests:
+    machine: true
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install --upgrade pip
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Pull images and setup
+          command: |
+            set -x
+            docker-compose up -d
+            sleep 10
```

Comment:
This feels brittle... did you try the following:

1. `docker-compose up -d --no-start` -> this will create all the necessary containers.
2. `docker-compose start postgres` -> this will return when Postgres is running.
3. Now you can proceed to creating the database and running the tests.

?

Reply:
That's pretty slick, I should give that a go since ```--rm``` will run the containers anyway.